src/dmd/target.d

```diff
@@ -1254,6 +1256,56 @@ struct TargetC
         }
     }
 
+    /***********************************
+     * Add global target-dependent builtin declarations.
+     * Params:
+     *  symbols = array to push constructed target built-ins
+     */
+    extern (C++) void addBuiltinDeclarations(Dsymbols *symbols) const
+    {
+        import dmd.astenums : VarArg;
+        import dmd.dstruct : StructDeclaration;
+        import dmd.globals : Loc;
+        import dmd.identifier : Identifier;
+        import dmd.id : Id;
```

Comment:
Having target.d dependent on everything else in the compiler is not the best.

Reply:
Or this could be part of compiler.d, but then that would be a new import required for cparse.

Reply:
It may not be the best but all of these are perfectly well understood by the other codebases, surely?

Anyway, working builtins could be a huge win for ImportC.

Reply:
`d-target.cc` includes most front-end headers, because it's ultimately got to translate things from DMD AST to GCC trees, then back into AST again on the fly in order to discover what information it should be providing to the DMD semantic routines.

It is impossible for Target to be a dumb interface, because - for example - you need a completed `Type` (lowered to a backend `tree` type) in order for the back-end to tell us how it will pass/return it, or how it will align it in a record, etc.

For this target hook, we are starting from the position where the back-end knows all the `__builtin_` functions available to use, but they are all in GCC tree format, and so there has to be a way to iterate through them all in order to build DMD AST nodes representing each one - internally referencing the original tree via the `csym` field - and push it into the importC compilation.